### ISSUE TITLE
naive use of styled-components' button component type breaks the rules of hooks

### ISSUE THREAD

**Issue by jfhbrook** [styled-components/styled-components#3016]
Hello! I tried to use styled-components for the first time and it blew up in my face! I did some digging and asking around online and as far as I can tell I'm not doing anything particularly wrong 😉 and my gut says that this is a styled-components but related to the particulars of my environment.

## Environment

```
(base) PS C:\Users\Josh\source\repos\jfhbrook\ihydra> npx envinfo --system --binaries --npmPackages styled-components,babel-plug
in-styled-components --markdown --clipboard
npx: installed 1 in 1.826s

*** Clipboard option removed - use clipboardy or clipboard-cli directly ***

## System:
 - OS: Windows 10 10.0.17763
 - CPU: (4) x64 Intel(R) Core(TM) i5-7500 CPU @ 3.40GHz
 - Memory: 6.62 GB / 15.96 GB
## Binaries:
 - Node: 12.6.0 - C:\Program Files\nodejs\node.EXE
 - Yarn: 1.16.0 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
 - npm: 6.12.1 - C:\Program Files\nodejs\npm.CMD
## npmPackages:
 - styled-components: ^5.0.1 => 5.0.1
```

(as an aside, the envinfo command needs to be updated to account for the --clipboard option being removed from npx)

## Reproduction

I unfortunately don't have a minimum repro, I'm pretty new to React and don't know what I'm looking for, but I do have an existing Electron app with a small change that reproduces. Here's the compare:

<https://github.com/jfhbrook/ihydra/compare/develop...jfhbrook:styled-button>

The issue I'm running into happens after this change, but everything's fine prior.

## Steps to reproduce

* Write an application using hooks :)
* Create a styled-components button component and add it to non-stateful components
* Try to use that component in the context of a hooks-using stateful container (or at least, my container)

## Expected Behavior

I expected the button component I was using prior (based on MUI) with something else that vaguely looked like a styled button.

## Actual Behavior

React throws a fit about the rules of hooks:

![image](https://user-images.githubusercontent.com/57536/74284837-968f6d00-4cf2-11ea-9874-f04fdc36573c.png)

Naively, one might think that I screwed up somehow, but I feel pretty good about the bulleted list this presents to me:

### 1. I only have one each of react and react-dom in my tree and the versions match on both:

```
(base) PS C:\Users\Josh\source\repos\jfhbrook\ihydra> npm ls react
ihydra@5.2.0 C:\Users\Josh\source\repos\jfhbrook\ihydra
`-- react@16.12.0

(base) PS C:\Users\Josh\source\repos\jfhbrook\ihydra> npm ls react-dom
ihydra@5.2.0 C:\Users\Josh\source\repos\jfhbrook\ihydra
`-- react-dom@16.12.0
```

### 2. I'm not calling Button as a non-component anywhere so it's not that kind of silly bug

```
(base) PS C:\Users\Josh\source\repos\jfhbrook\ihydra> ag Button
src/renderer/components/Alert.js
4:import Button from "./Button";
12:        <Button key={label} onClick={hook}>
14:        </Button>

src/renderer/components/Button.js
3:const Button = styled.button`
8:export default Button;

src/renderer/components/JupyterCommandFinder.js
9:import Button from "./Button";
41:      <Button onClick={selectFile}>
43:      </Button>
44:      <Button onClick={trySearching}>detect jupyter automatically</Button>
45:      <Button onClick={submit}>use this command</Button>
46:      <Button onClick={exit}>exit</Button>

src/renderer/components/JupyterRuntime.js
6:import Button from "./Button";
12:      <Button onClick={stopJupyter}>Stop Jupyter</Button>

src/renderer/components/MainMenu.js
5:import Button from "./Button";
18:      <Button onClick={tryInstall}>Install IHydra</Button>
19:      <Button onClick={goBackToWhich}>Find Jupyter</Button>
20:      <Button onClick={launchJupyter}>Launch Jupyter</Button>
21:      <Button onClick={exit}>Exit</Button>

src/renderer/components/StackTrace.js
4:import Button from "./Button";
13:      <Button onClick={retry}>Try Again</Button>
14:      <Button onClick={fail}>Exit</Button>
```

### 3. I've been working on this app for a few weeks now and every other time I've broken the rules of hooks it's turned out to be due to something I was doing, usually around catching exceptions related to component instantiation inside of my state functions. That isn't what's happening here - nothing's throwing an error in any of the components, it's just... breaking the rules of hooks.

I also flipped through the stack trace in the console error and it pointed at this react call with a mildly alarming comment?

![image](https://user-images.githubusercontent.com/57536/74285355-a9567180-4cf3-11ea-98aa-afb1abe8ba72.png)

Again, I'm sorry I don't have a minimal repro case. I'm hoping the notes I *do* have are helpful. If there are concrete things I can do to help debug this let me know. My app doesn't have great directions for setting it up right now, but if you're game `npm i`, `npm run rebuild` and `npm run dev` on that branch should get it reproducing locally.

Thanks!

**Comment by benjaminEwhite** [comment]
I'm getting same error, and am not explicitly using Hooks anywhere. I've used styled components in other projects and had no issue. This project I did not bootstrap with `create-react-app` so there's that difference, but I can narrowly isolate the React hook error to importing styled components into a component module file.

**Comment by benjaminEwhite** [comment]
I set my react, react-dom and styled-components versions to what I was using in another project, and works as expected. Specifically with:

- react@16.8.6
- react-dom@16.8.6
- styled-components@4.1.3

**Comment by chulander** [comment]
i opened and [issue](https://github.com/styled-components/styled-components/issues/3015) that was closed.

**Comment by benjaminEwhite** [comment]
FWIW, I ended up getting things to work with current version. In my case, I had misconfigured a multi-site setup, and the HTML files didn't actually have `<div id='root'></div>` anywhere like my code expected. Ultimately, the app was never getting attached to the DOM, even though this was unclear from the error message.

After correctly configuring webpack so that app was attaching to DOM, problem gone.
